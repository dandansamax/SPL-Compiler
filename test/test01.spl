int main()
{
    int a = 1;
    a = 1+1;
}