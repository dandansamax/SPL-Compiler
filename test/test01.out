FUNCTION mod :
PARAM var_4
PARAM var_5
t14 := var_4
t18 := var_4
t19 := var_5
t16 := t18 / t19
t17 := var_5
t15 := t16 * t17
t13 := t14 - t15
RETURN t13
FUNCTION gcd :
PARAM var_2
PARAM var_3
t6 := var_3
IF t6 != #0 GOTO L0
GOTO L1
LABEL L0 :
t8 := var_3
t10 := var_2
t11 := var_3
ARG t11
ARG t10
t9 := CALL mod
ARG t9
ARG t8
t7 := CALL gcd
RETURN t7
GOTO L2
LABEL L1 :
t12 := var_2
RETURN t12
LABEL L2 :
FUNCTION main :
READ t5
var_0 := t5
READ t4
var_1 := t4
t2 := var_0
t3 := var_1
ARG t3
ARG t2
t1 := CALL gcd
WRITE t1
t0 := #0
RETURN t0
